compiler/rustc_parse/src/lib.rs

```diff
@@ -200,16 +200,23 @@ pub fn maybe_file_to_stream(
         lexer::parse_token_trees(sess, src.as_str(), source_file.start_pos, override_span);
 
     match token_trees {
-        Ok(stream) => Ok((stream, unmatched_braces)),
-        Err(err) => {
+        Ok(stream) if unmatched_braces.is_empty() => Ok((stream, unmatched_braces)),
```

Comment:
This change and the EOF change are two distinct changes with different effects on compiler outputs.
Could you split them into two commits so their individual effects on tests are visible?

Reply:
Split out https://github.com/rust-lang/rust/pull/108297/commits/24117a32c92cd1b5d86b56250d1d72efa62c8789